ckanext/canada/view.py

```diff
@@ -395,10 +395,14 @@ def create_pd_record(owner_org, resource_name):
                         for (k, v) in ve.error_dict['records'][0].items()
                     }, **err)
                 except AttributeError:
-                    err = dict({
-                        k: [_("This record already exists")]
-                        for k in pk_fields
-                    }, **err)
+                    if 'duplicate key value violates unique constraint' in ve.error_dict['records'][0]:
+                        err = dict({
+                            k: [_("This record already exists")]
+                            for k in pk_fields
+                        }, **err)
+                    else:
+                        h.flash_error(_('Something went wrong, unable to create your record. Please contact support.'))
+                        err = 'OtherDatabaseError'
```

Comment:
@wardi looks like our templates were using the chromos to match the fields and not calling the form macro for errors. So I just added in a `error_summary` template variable and used that instead of the flash messages.